lib/tree.ml

```diff
@@ -69,7 +69,7 @@ let string_of_perm = function
   | `Normal -> "100644"
   | `Exec   -> "100755"
   | `Link   -> "120000"
-  | `Dir    -> "40000"
+  | `Dir    -> "040000"
```

Comment:
are you sure about this one? this means `perm_of_string (string_of_perm x)` becomes different of `x`.